app/ts/store/account.ts

```diff
@@ -1,77 +1,89 @@
-import { signal } from '@preact/signals'
+import { effect, Signal, useSignal } from '@preact/signals'
 import { ethers } from 'ethers'
-import { isEthereumJsonRpcError, EthereumJsonRpcError } from '../library/exceptions.js'
+import { AsyncProperty, useAsyncState } from '../library/preact-utilities.js'
 import { assertsExternalProvider, isEthereumObservable } from '../library/utilities.js'
 
-type AccountBusy = {
-	status: 'busy'
+export type ConnectMutation = {
+	transport: Signal<AsyncProperty<string>>
+	dispatch: () => void
+	reset: () => void
 }
 
-type AccountConnected = {
-	status: 'connected'
-	address: string
+export type ReconnectMutation = {
+	transport: Signal<AsyncProperty<unknown>>
+	dispatch: () => void
+	reset: () => void
 }
 
-type AccountDisconnected = {
-	status: 'disconnected'
-	connect: () => Promise<void>
-	ensureConnected: () => Promise<void>
-}
+type Account =
+	| {
+			isConnected: false
+			connectMutation: ConnectMutation
+			reconnectMutation: ReconnectMutation
+	  }
+	| {
+			isConnected: true
+			address: string
+	  }
 
-type AccountConnectRejected = {
-	status: 'rejected'
-	error: EthereumJsonRpcError | Error
-}
+export type AccountStore = ReturnType<typeof createAccountStore>
+export function createAccountStore() {
+	const { value: asyncValue, waitFor, reset } = useAsyncState<string>()
+	const reconnect = useAsyncState()
+
+	const reconnectMutation = {
+		transport: reconnect.value,
+		dispatch: () => {
+			reconnect.waitFor(async () => {
+				try {
+					assertsExternalProvider(window.ethereum)
+					const provider = new ethers.providers.Web3Provider(window.ethereum)
+					const signer = provider.getSigner()
+					const address = await signer.getAddress()
+					accountStore.value = { isConnected: true, address }
+				} catch {
+					accountStore.value = accountDefaults
+				}
+			})
+		},
+		reset: reconnect.reset,
+	}
 
-export type AccountStore = AccountBusy | AccountConnected | AccountDisconnected | AccountConnectRejected
+	const connectMutation = {
+		transport: asyncValue,
+		dispatch: () => {
+			waitFor(async () => {
+				assertsExternalProvider(window.ethereum)
+				const provider = new ethers.providers.Web3Provider(window.ethereum)
+				await provider.send('eth_requestAccounts', [])
+				const signer = provider.getSigner()
+				const address = await signer.getAddress()
+				accountStore.value = { isConnected: true, address }
+				return address
+			})
```

Comment:
Consider removing the excess curlies here.
```suggestion
		dispatch: () => waitFor(async () => {
			assertsExternalProvider(window.ethereum)
			const provider = new ethers.providers.Web3Provider(window.ethereum)
			await provider.send('eth_requestAccounts', [])
			const signer = provider.getSigner()
			const address = await signer.getAddress()
			accountStore.value = { isConnected: true, address }
			return address
		)
```